lib/components/CelebGallery/index.tsx

```diff
@@ -1,52 +1,56 @@
 import React from 'react';
 import Image from 'next/image';
 import { sanityImage } from '~/lib/components/sanityio';
-import { Typography } from '@mui/material';
-import s from './styles.module.scss';
 import { TCelebGalleryItem } from '~/lib/components/types';
 import { useCelebContext } from '~/lib/components/StaticPropsContextProvider';
+import {useRouter} from 'next/router';
 
 export const CelebGallery: React.FC<{
   celebGalleryItems: TCelebGalleryItem[];
 }> = (p) => {
   const context = useCelebContext();
+  const router = useRouter()
 
+  // dynamic classes
+  let grid_class = ['grid', 'gap-4', router.pathname == '/' ? 'grid-cols-2' : 'grid-cols-3', router.pathname == '/' ? 'sm:grid-cols-3' : 'sm:grid-cols-4', 'lg:grid-cols-4', 'items-center', 'justify-center', 'w-full'].join(' ')
   return (
-    <div className={s.CelebGallery}>
-      {p.celebGalleryItems.map((celebData) => {
-        const picture = celebData.picture || context.placeholderImage;
+    <div className="container mx-auto w-full h-auto items-center justify-center">
+      <div className={grid_class}>
+        {p.celebGalleryItems.map((celebData) => {
+          const picture = celebData.picture || context.placeholderImage;
 
-        return (
-          <a
-            className={s.link}
-            href={`/${celebData.slug}`}
-            key={celebData.slug}
-          >
-            <span className={s.image}>
-              <Image
-                objectFit="cover"
-                objectPosition="top"
-                blurDataURL={picture.metadata.lqip}
-                placeholder="blur"
-                src={sanityImage(picture).width(200).height(250).url()}
-                alt={celebData.name}
-                layout="fixed"
-                width={160}
-                height={200}
-                className={s.image}
-              />
-            </span>
-            <Typography
-              fontWeight={500}
-              variant="h4"
-              component="p"
-              className={s.name}
+          return (
+            <a
+              className="celeb-card overflow-hidden rounded-xl w-full"
+              href={`/${celebData.slug}`}
+              key={celebData.slug}
             >
-              {celebData.name}
-            </Typography>
-          </a>
-        );
-      })}
+              <span className="relative block w-full h-auto">
+                <Image
+                  objectFit="cover"
+                  objectPosition="center"
+                  blurDataURL={picture.metadata.lqip}
+                  placeholder="blur"
+                  src={sanityImage(picture).url()}
+                  alt={celebData.name}
+                  layout="responsive"
+                  width={200}
+                  height={225}
+                  className="rounded-lg w-full h-auto block"
```

Comment:
I see `w-full`, `h-auto`, and `block` being used here. But are they really necessary?

Can you also please review the other classes that you are using and make sure we don't have unnecessary classes.